### ISSUE TITLE
Index Checker: remove common obvious false positives( array access: constant index) suggestion.

### ISSUE THREAD

**Issue by NhatDinh** [typetools/checker-framework#1905]
Consider this code block:
```java
import org.checkerframework.checker.index.qual.IndexFor;
import org.checkerframework.common.value.qual.MinLen;

public class SimpleMethod1 {
   double meanOf(@MinLen(1) int[] values,@MinLen(10) double[] double_values ) {
       double double_mean = double_values[5];
       int mean = values[0];
       return mean;
   }
   public static void main() {
   }
}
```
Warnings received:
```java
Error:(6, 44) java: [array.access.unsafe.high.constant] Potentially unsafe array access: the constant index 5 could be larger than the array's bound
  found   : @ArrayLenRange(from=10, to=2147483647) double @UnknownVal []
  required: @MinLen("6") -- an array guaranteed to have at least 6 elements
Error:(7, 27) java: [array.access.unsafe.high.constant] Potentially unsafe array access: the constant index 0 could be larger than the array's bound
  found   : @ArrayLenRange(from=1, to=2147483647) int @UnknownVal []
  required: @MinLen("1") -- an array guaranteed to have at least 1 elements
```
Warnings are two very obvious false positive cases, array access at constant index 5 could never be larger than @ArrayLenRange(from=10, to=2147483647), same with constant index 0 and @MinLen("1"). 
Reason: Remove these common false positive will resolve in programmers not having to write two more @SupressWarnings
Proposed solution: A solution to this might be to compare the constant index with argument passed in @MinLen(int min) or the found @ArrayLenRange(from=min), if index is less than min, no warning should be prompted.

**Comment by mernst** [comment]
The problem here is incorrect syntax.
`@MinLen(10) double[]` means an array of `@MinLen(10) double`.
You instead want to write `double @MinLen(10) []` which means a `@MinLen(10)` array of double.

See section 2.1 in the manual, ["Writing annotations"](https://checkerframework.org/manual/#writing-annotations).

(It ought to be illegal to write `@MinLen` on a primitive, but the checker doesn't warn about that yet.)

**Comment by NhatDinh** [comment]
Thank you for pointing that out. I won't make that mistake again.

**Comment by mernst** [comment]
It's not a problem.  The fact that the Index Checker doesn't issue a warning when `@MinLen` is applied to a primitive type is a bug, and would be worth fixing.

**Comment by NhatDinh** [comment]
If this is a worthwhile bug, I would like to work on fixing it.

**Comment by mernst** [comment]
That is fine with me.  Thanks!
